src/components/utils/hooks/useXmtpClient.tsx

```diff
@@ -5,6 +5,19 @@ import { useAppStore } from 'src/store/app';
 import { useMessageStore } from 'src/store/message';
 import { useSigner } from 'wagmi';
 
+const ENCODING = 'binary';
+
+const buildLocalStorageKey = (walletAddress: string) => `xmtp:keys:${walletAddress}`;
+
+const loadKeys = (walletAddress: string): Uint8Array | null => {
+  const val = localStorage.getItem(buildLocalStorageKey(walletAddress));
+  return val ? Buffer.from(val, ENCODING) : null;
+};
+
+const storeKeys = (walletAddress: string, keys: Uint8Array) => {
+  localStorage.setItem(buildLocalStorageKey(walletAddress), Buffer.from(keys).toString(ENCODING));
```

Comment:
Can we clear the localStorage on logout? I think currently we're listening for changes in `currentProfile` to determine whether to reset the state or not in other hooks.